packages/core/src/storage/vectorStore/ChromaVectorStore.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  AddParams,
+  ChromaClient,
+  Collection,
+  Embeddings,
+  IncludeEnum,
+  Where,
+  WhereDocument,
+} from "chromadb";
+import { BaseNode, Document, MetadataMode } from "../../Node";
+import {
+  VectorStore,
+  VectorStoreQuery,
+  VectorStoreQueryMode,
+  VectorStoreQueryResult,
+} from "./types";
+
+type ChromaDBParams = {
+  collectionName: string;
+};
+
+type ChromaDeleteOptions = {
+  where?: Where;
+  whereDocument?: WhereDocument;
+};
+
+type ChromaQueryOptions = {
+  whereDocument?: WhereDocument;
+};
+
+export class ChromaVectorStore implements VectorStore {
+  storesText: boolean = true;
+  private chromaClient: ChromaClient;
+  private collection: Collection | null = null;
+
+  constructor() {
+    this.clearCollection();
+    this.chromaClient = new ChromaClient();
+  }
+
+  client(): ChromaClient {
+    return this.chromaClient;
+  }
+
+  async setCollection(collectionName: string) {
+    const coll = await this.chromaClient.createCollection({
+      name: collectionName,
+    });
+    this.collection = coll;
+  }
+
+  getCollection(): Collection {
+    if (!this.collection) {
+      throw new Error("Must initialize collection before getting.");
+    }
+    return this.collection;
+  }
+
+  clearCollection(): void {
+    this.collection = null;
+  }
+
+  private getDataToInsert(nodes: BaseNode[]): AddParams {
+    return {
+      embeddings: nodes.map((node) => node.getEmbedding()),
+      ids: nodes.map((node) => node.id_),
+      metadatas: nodes.map((node) => node.metadata),
+      documents: nodes.map((node) => node.getContent(MetadataMode.NONE)),
+    };
+  }
+
+  async add(nodes: BaseNode[]): Promise<string[]> {
+    if (!nodes || nodes.length === 0) {
+      return [];
+    }
+
+    const dataToInsert = this.getDataToInsert(nodes);
+
+    try {
+      await this.getCollection().add(dataToInsert);
+    } catch (err) {
+      const msg = `${err}`;
+      console.log(msg, err);
+      return [];
+    }
```

Comment:
how about we just do this:
```suggestion
    await this.getCollection().add(dataToInsert);
```
then a user can log an error however they want (same for the other methods)